libsql-server/src/namespace/meta_store.rs

```diff
@@ -41,10 +45,26 @@ pub struct MetaStoreHandle {
 
 #[derive(Debug, Clone)]
 enum HandleState {
-    Internal(Arc<Mutex<Arc<DatabaseConfig>>>),
+    Internal(Arc<Mutex<MetaState>>),
     External(mpsc::Sender<ChangeMsg>, Receiver<InnerConfig>),
 }
 
+#[derive(Debug, Default)]
+struct MetaState {
+    config: Arc<DatabaseConfig>,
+    shared_schemas: HashMap<NamespaceName, HashSet<NamespaceName>>,
```

Comment:
It's a map from shared_schema -> set of dbs that use that shared schema.

Reply:
it's not the place to store that. One instance of that is created for every namespace. Why do we need that mapping in memory?

Reply:
This will be empty for anything but shared schema database

Reply:
i still don't understand why this needs to me in memory. Also, why is it is mapping namespaces to maps of namespaces? Are we storing all the shared schema databases to the databases that uses them? Why?

Reply:
it's a mapping from shared schema S to set of namespaces that use S.
Remember that shared schema is just a special namepace so
HashMap<NamespaceName, HashSet< NamespaceName>>
       shared schema S -^ ...............  ^-------------------------^--------- dbs using S

Reply:
that's one thing yes, but still, I don't understand why this is cached at all?

Reply:
I'm not sure I understand the question. This is cached for speed :). The same way DatabaseConfig is I guess.

Reply:
yes, but:
1) the shared schema is almost never accessed, it's not like you are doing 10 migrations per second
2) it can get very big, if you have 100s of thousands of dbs for a single namespace

Reply:
Those are fair points. So instead of caching in memory we fetch the list from the meta store each time it's being used. That's what you're proposing?

Reply:
yeah, might even be an iterator, it we don't need the whole list at once

Reply:
@MarinPostma @haaawk for the context: this piece of code is made for compliance with `HandleState::Internal` which - from what I've seen - is used only for testing purposes.